app/scripts/views/sign_up.js

```diff
@@ -199,6 +201,10 @@ function (Cocktail, _, p, BaseView, FormView, Template, AuthErrors, mailcheck,
         });
     },
 
+    passwordCheck: function () {
+      passwordcheck(this, this.$el.find('#password'), this.metrics, this.translator);
```

Comment:
Woah, this passes in metrics and the translator to passwordcheck. What happens there?

I'm going to suggest an alternative API so that passwordcheck can be used in projects other than our own:

```js
passwordcheck(<string:password>, <function:onWeakPassword>)
```

Pass a password and a callback into passwordcheck. Whenever there is a match, call the callback. Inside the callback, register an event with metrics.

This will also allow each project to show their own custom UI when there is a weak password.

```js
var self = this;
var password = this.getElementValue('.password');
passwordcheck(password, function () {
  self.logScreenEvent('weak_password');
  // in the future, do some fancy tooltip here.
});
```

This way, the password checking logic is completely decoupled from the view/logging logic. Both are free to be updated as long as the API to passwordcheck remains the same. passwordcheck makes no assumptions about its environment, and could even be used by a Node backend.